### PR TITLE
Feat: add some dubbo config and use new config style

### LINKED ISSUE #631: [feature] Support more custom config for Dubbo Client
**What would you like to be added**:
Support more custom config parameters for Dubbo Client

Same as #615 #625 

**Why is this needed**:

### PR DESCRIPTION
**Pull request by Similarityoung**
### **What this PR does**:

This pull request significantly refactors and extends the Dubbo client configuration and integration logic to support more flexible and feature-rich Dubbo proxying. The main changes include expanding the `DubboProxyConfig` struct with new fields and helpers, updating the Dubbo client creation to use the latest dubbo-go v3.3.1 client API, and adding comprehensive unit tests for the new configuration options.

Key changes:

**DubboProxyConfig Enhancements**
* Added new fields to `DubboProxyConfig` for advanced Dubbo features: `Cluster`, `Check`, `Protocol`, `Filter`, `Serialization`, `Sticky`, and `Params`, each with documentation and sensible defaulting logic. Helper methods (`GetCluster`, `GetProtocol`, `GetCheck`) were introduced to provide default values and maintain backward compatibility.

**Dubbo Client Refactor**
* Switched Dubbo client initialization from the old `dg.RootConfig` approach to the new `dclient.NewClient` API, updating the registry and application configuration handling to use `global.RegistryConfig` and `global.ApplicationConfig`. [[1]](diffhunk://#diff-1f73347753bf43f97bf0ecd965437bb343d893ca0884e7707ddbe40aaef841cbR23-L34) [[2]](diffhunk://#diff-1f73347753bf43f97bf0ecd965437bb343d893ca0884e7707ddbe40aaef841cbL84-R85) [[3]](diffhunk://#diff-1f73347753bf43f97bf0ecd965437bb343d893ca0884e7707ddbe40aaef841cbL98-R100) [[4]](diffhunk://#diff-1f73347753bf43f97bf0ecd965437bb343d893ca0884e7707ddbe40aaef841cbL136-R168)
* Refactored the service reference creation logic to use a new `buildReferenceOptions` method, which builds the reference configuration using the new API and applies all advanced options from `DubboProxyConfig` and the integration request. This enables support for all new configuration fields and ensures correct option precedence.

**Testing**
* Added comprehensive unit tests in `config_test.go` to verify the new configuration fields and helper methods, including edge cases for defaulting and explicit configuration.

### **Which issue(s) this PR fixes**:

Fixes #631

### CODE REVIEW

File: tools/benchmark/go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/apache/dubbo-go-pixiu/tools/benchmark
 go 1.25.0
 
 require (
-	dubbo.apache.org/dubbo-go/v3 v3.3.1
+	dubbo.apache.org/dubbo-go/v3 v3.0.0-20260210015753-35ea886421f9
```
Comment (Alanxtl) [inline tools/benchmark/go.mod:6]: 为啥还给降级了

Comment (Similarityoung) [reply tools/benchmark/go.mod:6]: 那啥，这个 tag 的版本比 3.3.1 要高，可以用新版的泛化调用 api，主要是等 dubbo-go 那发 3.3.2

File: pkg/client/dubbo/dubbo.go [hunk 2]
```diff
@@ -172,106 +196,157 @@ func (dc *Client) Close() error {
...
+		Retries:        "3",
+		RequestTimeout: cst.DefaultReqTimeout,
+	}
+
+	if dc.dubboProxyConfig == nil {
+		return defaults
+	}
+
+	defaults.LoadBalance = dc.dubboProxyConfig.LoadBalance
+	if strings.TrimSpace(dc.dubboProxyConfig.Retries) != "" {
```
Comment (AlexStocks) [inline pkg/client/dubbo/dubbo.go:288]: [P1] 这里把 retry 来源收窄成全局 DubboProxyConfig.Retries，但 IntegrationRequest.DubboBackendConfig.Retries 字段仍保留，现有示例 docs/sample/dubbo/dubbo-multi.md 还在按单个 API 配置 retries。旧实现会在全局 retries 为空时回退到 irequest.Retries；现在 DubboHandler 没有把 ir.Retries 放进 DubboOutboundRequest，resolvedReferSpec 也没有这个字段，最终 WithRetries 只能拿到全局默认值或全局配置。结果是按 API 配的 retries 静默失效。需要把 per-route retries 传到 outbound/spec 并在 buildReferenceOptions 中按旧优先级处理，或删除字段和文档并在配置阶段显式拒绝该配置。

File: pkg/filter/http/remote/call.go [hunk 3]
```diff
@@ -145,25 +131,27 @@ func (f *Filter) Decode(c *contexthttp.HttpContext) filter.FilterStatus {
 	}
 
 	typ := api.IntegrationRequest.RequestType
+	switch strings.ToLower(typ) {
```
Comment (AlexStocks) [inline pkg/filter/http/remote/call.go:137]: [P1] 这里直接进入静态 API 的 requestType 分发，PR 同时删除了 DubboProxyConfig.AutoResolve 和 resolver 逻辑。已有配置如果设置 auto_resolve: true，当前结构体不再接收该字段，运行时也不会再按请求路径动态解析 application/interface/method，而是只使用已匹配的静态 API。这个外部配置兼容性破坏没有迁移或显式报错，旧配置会静默变成另一套路由行为。需要保留兼容路径，或在配置加载阶段检测并拒绝 auto_resolve，同时补充迁移文档和测试。

File: pkg/client/dubbo/dubbo.go (hunk 2 above)
Comment (Similarityoung) [reply pkg/client/dubbo/dubbo.go:288]: 雨哥觉得需要对每个 api 配置 retry 吗，我目前想只需要全局的 retry 而不对每个单独的 api 设置，因为构造dubbo client 时，如果retry 不同会导致制造不同的 client，而无法复用之前创建好的 client。

File: pkg/filter/http/remote/call.go (hunk 3 above)
Comment (Similarityoung) [reply pkg/filter/http/remote/call.go:137]: 已经检测并拒绝 auto_resolve，同时补充迁移文档和测试。

**Review by Alanxtl — APPROVED** [review]
lgtm
